api/envoy/extensions/filters/network/http_connection_manager/v4alpha/http_connection_manager.proto

```diff
@@ -765,14 +766,23 @@ message HttpFilter {
 
   reserved "config";
 
-  // The name of the filter to instantiate. The name must match a
-  // :ref:`supported filter <config_http_filters>`.
+  // The name of the filter to instantiate. The name is used as a fallback to
+  // select an extension if the type of the configuration proto is not
+  // sufficient.
   string name = 1 [(validate.rules).string = {min_bytes: 1}];
 
   // Filter specific configuration which depends on the filter being instantiated. See the supported
   // filters for further documentation.
   oneof config_type {
     google.protobuf.Any typed_config = 4;
+
+    // [#not-implemented-hide:] Configuration source specifier for FDS.
```

Comment:
Can we flesh out the docs here so we can agree on the behavior? At a high level something like:

> Filter configuration updates are atomic. They will be applied to new streams while existing streams will use previous configuration.

Additionally, I think we need this to have more configuration probably inside a message that allows for optional default config if the xDS server timeouts out / can't be reached. We also need to think through the behavior if no optional default config is specified. Are requests immediately failed? Do we always require default config? Etc.

Reply:
I was thinking of putting the listener into warming before HCM receives all the filter configs. That means the server should probably always send something (similar to RDS, to prevent stalling the config pipeline).

Reply:
Yes, it has to warm, but the server may fail and/or timeout. I would like to have at least some optional default config to guard against this (for example a WAF filter default config could be no blocking rules).

Reply:
Added default config option.